lib/frame.rb

```diff
@@ -0,0 +1,93 @@
+class Frame
+  FIRST_FRAME = 1
+  SECOND_FRAME = 2
+  STRIKE = 10
+  STRIKE_BONUS = 20
+
+  attr_reader :frames, :current_frame, :next_frame
+
+  def initialize(current_frame = FIRST_FRAME, next_frame = SECOND_FRAME)
+    @frames = (1..10).map { |x| [x, []] }.to_h
+    @current_frame = current_frame
+    @next_frame = next_frame
+  end
+
+  def add_first_roll(first_pin) 
+    if @current_frame > 1 && strike? 
+      if first_pin == STRIKE
+        consecutive_strike(first_pin)
+      else
+        @frames[@current_frame] = [first_pin]
+      end
+    elsif @current_frame > 1 && spare? 
+      if first_pin == STRIKE
+        spare_and_strike(first_pin)
+      else
+        @frames[@current_frame] = [first_pin]
+        @frames[@current_frame - 1][1] += first_pin
+      end
+    else
+      if first_pin == STRIKE
+      single_strike(first_pin)
+      else
+        @frames[@current_frame] = [first_pin]
+      end
+    end
+  end
```

Comment:
Bigoli code?

Reply:
Totally! A new level of spaghetti code! XD